src/hooks/coreSDK/safeCoreSDK.ts

```diff
@@ -39,19 +41,38 @@ export const createEthersAdapter = (provider = getWeb3()) => {
 }
 
 // Safe Core SDK
-export const initSafeSDK = async (
-  provider: EIP1193Provider,
-  chainId: string,
-  safeAddress: string,
-  safeVersion: string,
-): Promise<Safe> => {
+export const initSafeSDK = async (provider: EIP1193Provider, safe: SafeInfo): Promise<Safe | undefined> => {
+  const masterCopy = safe.implementation.value
+
+  const chainId = safe.chainId
+  const safeAddress = safe.address.value
+  let safeVersion = safe.version
+
   let isL1SafeMasterCopy = chainId === chains.eth
-  // Legacy Safe contracts
-  if (isLegacyVersion(safeVersion)) {
-    isL1SafeMasterCopy = true
-  }
 
   const ethersProvider = new Web3Provider(provider)
+
+  // If it is an official deployed master copy we should still initiate the safeSDK
+  if (safeVersion === null) {
+    safeVersion = await Gnosis_safe__factory.connect(safeAddress, ethersProvider).VERSION()
+
+    const safeL1Deployment = getSafeSingletonDeployment({ network: chainId, version: safeVersion })
+    const safeL2Deployment = getSafeL2SingletonDeployment({ network: chainId, version: safeVersion })
+
+    isL1SafeMasterCopy = masterCopy === safeL1Deployment?.defaultAddress
+    const isL2SafeMasterCopy = masterCopy === safeL2Deployment?.defaultAddress
+
+    if (!isL1SafeMasterCopy && !isL2SafeMasterCopy) {
+      // Unknown masterCopy, which we do not want to support
+      return Promise.resolve(undefined)
+    }
+  } else {
```

Comment:
Afaiu the main fix in this PR is that we now initialize the Core SDK even if a `safe.version` doesn't exist but I don't understand why we need this part.

To test, I deployed a Safe on Polygon with the L1 master copy contract instead of L2 (`matic:0x2B7a8C8afaEc010140970488190275A26b1436DD`) which should fall under the "unsupported contract" category. For some reason, the backend still returns a version for this Safe so it is not caught by this condition but it is still possible to create and execute transactions.

Should we instead use the same condition we use for showing the notification `isValidMasterCopy`?

Reply:
`isValidMasterCopy` seems to catch Safes with an unsupported contract both when there is a version and when there is no version.